go/logic/migrator.go

```diff
@@ -396,6 +396,10 @@ func (this *Migrator) handleCutOverResult(cutOverError error) (err error) {
 	// Only on error:
 
 	if this.migrationContext.TestOnReplica {
+		if err := this.hooksExecutor.onTestReplicaComplete(); err != nil {
```

Comment:
My apologies! I thought #406 and the description conveyed what I was after however it doesn't appear I did 😞  The intention was to have a hook that could be used with `--test-on-replica` runs and would run before the cut over would happen (and not on all migrations, just the test ones) to do things like integrity checks + checksumming whilst the replication was stopped and not yet complete.

> We use it that way, and just pass completely different hooks to the test-on-replica invocation.

I did explore this but could have been doing it wrong. What do you use to differentiate between the two types of migrations? Would you mind sharing a simple example of how you're achieving this?